### DISCUSSION TITLE
Renaming imports

### DISCUSSION THREAD

**Discussion by last-partizan** [microsoft/pylance-release#4409] [Enhancement]
Hello, i'm using `pyright` as language server with neovim, so this issue is more related to pyright, but i was advised to create feature request here (https://github.com/microsoft/pyright/issues/5148).

When i try to do the same with vscode+pylance, result is exactly the same.

-----

Today i was trying to rename imported name, but instead pyright renamed a class.

```
from app.models import Book
```

I tried to rename it to `SpecialBook`, and instead of changing import to `Book as SpecialBook`, pyright renamed my target class.

**Describe the solution you'd like**
I would like it to change import, not target class.

**Additional context**
This would be expected behaviour, becouse other LSP servers (like volar for vue) are using similar approach.

Do you think it's possible to fix this in pyright renaming logic, and not to make a pylance-only feature?

**Comment by debonte** [comment]
Moving this issue to discussions as an enhancement request for comments and upvotes.
